roofit/roofitcore/src/FitHelpers.cxx

```diff
@@ -111,11 +152,11 @@ int calcAsymptoticCorrectedCovariance(RooAbsReal &pdf, RooMinimizer &minimizer,
          *paraminternal = paramresult->getVal();
          diffs[k] = diff;
       }
+
       // Fill numerator matrix
-      double prob = pdf.getVal(&obs);
       for (std::size_t k = 0; k < floated.size(); k++) {
          for (std::size_t l = 0; l < floated.size(); l++) {
-            num(k, l) += data.weightSquared() * diffs[k] * diffs[l] / (prob * prob);
+            num(k, l) += data.weightSquared() * (diffs[k] + diffs_expected[k]) * (diffs[l] + diffs_expected[l]);
```

Comment:
Hi @langenbruch! To you somewhere have a paper to quote for this formula, or some notes? I can't find it in https://arxiv.org/abs/1911.01303, which is currently the reference cited in the [RooFit documentation](https://arxiv.org/abs/1911.01303).

Reply:
Dear @guitargeek, the approach is exactly the same as in my paper, performing a Taylor expansion of the logarithmic likelihood (see Sec. 2.1), and then calculating the appropriate expectation values. Since the calculation for the extended term may not be completely straightforward, I wrote down the explicit steps in the attached PDF.  
[extended_weighted_fits.pdf](https://github.com/root-project/root/files/14663828/extended_weighted_fits.pdf)

Reply:
Thank you so much for the effort in writing up the document with the calculations! We should make the document more visible from the ROOT documentation, in case someone is interested in understanding the details. Would that be okay for you?

I don't think GitHub is not the best place to upload this. Can we put it on the ROOT website in some folder to link from the RooAbsPdf doxygen? Or you would prefer something else?